### PR TITLE
Add optional setup function, update Q dependency, bump version

### PR DESCRIPTION
**Pull request by mikkot**
Hello @nicks, @koop, 

Please review the following commits I made in branch 'mikko-add-setup'.

fb23452b971cf48ac72b698503f20d291403f3e2 (2016-03-24 14:07:37 -0700)
Add optional setup function, update Q dependency, bump version

R=@nicks
R=@koop

### CODE REVIEW

**Comment by nicks** [comment]
lgtm

i wonder if we'll ever want setupFn to return a promise

**Comment by mikkot** [comment]
Excellent idea - will give it a go @nicks.

File: lib/testcase.js [hunk 1]
```diff
@@ -185,7 +190,9 @@ TestCase.prototype._run = function () {
   if (this._payload) req.write(this._payload)
   req.end()
 
-  var promise = defer.promise
+  var promise = setupPromise.then(function () {
```
Comment (nicks) [inline lib/testcase.js]: did you intend to do the setup and the http request in parallel? i could go either way on this

**Comment by nicks** [comment]
lg

File: lib/testcase.js (hunk 1 above)
Comment (mikkot) [inline lib/testcase.js]: I did not, and that is a bug. I missed the fact that the HTTP request fires off immediately even if the code extracts the promise from it later. -> Round 3.

**Comment by nicks** [comment]
lg

**Pull request merged**